### PR TITLE
python3Packages.sqids: init at 0.3.0

### PR DESCRIPTION
**Pull request by panicgh**
## Description of changes

https://sqids.org/python:
> Sqids (pronounced "squids") is an open-source library that lets you generate YouTube-looking IDs from numbers. These IDs are short, can be generated from a custom alphabet and are guaranteed to be collision-free.

Sqids was formerly known as Hashids.

The old https://hashids.org now redirects to https://sqids.org/?hashids

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

----

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.sqids</li>
    <li>python310Packages.sqids.dist</li>
    <li>python311Packages.sqids</li>
    <li>python311Packages.sqids.dist</li>
  </ul>
</details>

### CODE REVIEW

**Comment by panicgh** [comment]
@ofborg build python311Packages.sqid python310Packages.sqid

**Comment by panicgh** [comment]
@ofborg build python311Packages.sqids python310Packages.sqids

File: pkgs/development/python-modules/sqids/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    inherit pname version;
+    hash = "sha256-qAY41kOp0m+mua/4bvVwuDW5p0EpwY675Ux3W1JsqbE=";
+  };
+
+  nativeBuildInputs = [
+    setuptools-scm
+  ];
+
+  doCheck = true;
+
```
Comment (natsukium) [inline pkgs/development/python-modules/sqids/default.nix:26]:
```suggestion
```
In python modules, `installCheckPhase` is always executed unless `doCheck = false`.

**Review by natsukium — APPROVED** [review]
I left trivial comments, but I don't want to request a resubmission just for that change.
LGTM

File: pkgs/development/python-modules/sqids/default.nix (hunk 1 above)
Comment (panicgh) [reply pkgs/development/python-modules/sqids/default.nix:26]: Thanks, I will apply your suggestions next time when there is an upstream update.

Comment (panicgh) [reply pkgs/development/python-modules/sqids/default.nix:26]: Fixed in #263651.

**Pull request merged**